### PR TITLE
Small tweaks to sampling from model in vae

### PR DESCRIPTION
**Pull request by jpchen**
Uses the poutine mechanism to extract the model latent, instead of actually running the decoder

it's certainly fewer lines of code, though more machinery to work with to extract something you can do with a helper function. not sure what the 'right' approach is

### CODE REVIEW

File: examples/vae/utils/vae_plots.py [hunk 1]
```diff
@@ -5,16 +5,19 @@ def plot_conditional_samples_ssvae(ssvae, visdom_session):
     """
     This is a method to do conditional sampling in visdom
     """
+    from pyro import poutine
```
Comment (fritzo) [inline examples/vae/utils/vae_plots.py]: Why not move these imports to the top?

**Comment by jpchen** [comment]
> return the bernoulli probs as one of its return values 

yeah i like that better, though it sort of violates the idea of model taking parameters and returning data

**Comment by martinjankowiak** [comment]
there is no such contract

**Comment by jpchen** [comment]
it's the right way to think about generative models imo. in any case, i updated the PR to reflect that

File: examples/vae/utils/vae_plots.py [hunk 2]
```diff
@@ -1,3 +1,10 @@
+import matplotlib.pyplot as plt
```
Comment (fritzo) [inline examples/vae/utils/vae_plots.py]: This breaks the `.use('Agg')` behavior (as discussed offline)

**Pull request merged**